Zend/zend_operators.c

```diff
@@ -60,8 +60,8 @@ static _locale_t current_locale = NULL;
 /* Common code for SSE2 accelerated character case conversion */
 
 #define BLOCKCONV_INIT_RANGE(start, end) \
-	const __m128i blconv_start_minus_1 = _mm_set1_epi8((start) - 1); \
-	const __m128i blconv_end_plus_1 = _mm_set1_epi8((end) + 1);
+	const __m128i blconv_offset = _mm_set1_epi8((char)(-128 - start)); \
```

Comment:
Won't this underflow, causing undefined behavior? Or is `char` somehow exempt from this rule? `UBSAN` doesn't seem to catch this.

Reply:
Also, maybe `CHAR_MIN` for clarity?

Reply:
@iluuu1994 Underflow is needed and intentional here. I am not sure about the rules for UB and would appreciate clarification on that from anyone who knows C better than I.

Reply:
That may depend on whether `char` is defined as `signed char` or `unsigned char`. The C standard allows either. I suggest to be explicit here.

Reply:
Thanks for the suggestion about using `CHAR_MIN`. I think that should be OK.

Reply:
@cmb69 Interestingly `UBSAN` doesn't complain even with explicit `signed char`, which is why I was confused.

Reply:
@nielsdos That makes sense.